sourcegen-generator/src/main/java/io/micronaut/sourcegen/generator/visitors/UtilsAnnotationVisitor.java

```diff
@@ -0,0 +1,406 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.sourcegen.generator.visitors;
+
+import io.micronaut.core.annotation.*;
+import io.micronaut.inject.ast.ClassElement;
+import io.micronaut.inject.ast.PropertyElement;
+import io.micronaut.inject.processing.ProcessingException;
+import io.micronaut.inject.visitor.TypeElementVisitor;
+import io.micronaut.inject.visitor.VisitorContext;
+import io.micronaut.sourcegen.annotations.Equals;
+import io.micronaut.sourcegen.annotations.HashCode;
+import io.micronaut.sourcegen.annotations.Secret;
+import io.micronaut.sourcegen.annotations.ToString;
+import io.micronaut.sourcegen.generator.SourceGenerator;
+import io.micronaut.sourcegen.generator.SourceGenerators;
+import io.micronaut.sourcegen.model.*;
+
+import java.util.*;
+import javax.lang.model.element.Modifier;
+
+/**
+ * The visitor that generates the Utils class of a bean.
+ * The Utils class can have functions substituting toString, equals, and hashcode.
+ * However, each method needs to be annotated to be generated.
+ *      {@link ToString} annotation for toString function
+ *      {@link Equals} annotation for equals function
+ *      {@link HashCode} annotation for hashCode function
+ *
+ * @author Elif Kurtay
+ * @since 1.3
+ */
+
+@Internal
+public final class UtilsAnnotationVisitor implements TypeElementVisitor<Object, Object> {
```

Comment:
It feels like this implementation is doing too much. Ideally it would be split into separate visitors and add the annotation type to the generic arguments. So `EqualsAndHashCodeVisitor implements TypeElementVisitor<EqualsHashCode, Object>`, `ToStringVisitor implements TypeElementVisitor<ToString, Object>` etc. however I am aware that will require generating multiple Utils classes so maybe not ideal

Either way you need to override `getSupportedAnnotationNames()` and list the annotations this visitor processes.